libass/ass.c

```diff
@@ -893,8 +895,8 @@ static int process_fonts_line(ASS_Track *track, char *str)
         }
         track->parser_priv->fontname = strdup(p);
         ass_msg(track->library, MSGL_V, "Fontname: %s",
-               track->parser_priv->fontname);
-        return 0;
+               track->parser_priv->fontname ? track->parser_priv->fontname : "null");
```

Comment:
This might be confusing, but not sure if important. Perhaps print a nomem error _instead_ of this message?

Also, if the `strdup` fails for `Fontname`, won’t libass print “Not understood” for _each_ of the following lines? Can we avoid this somehow?

Reply:
> Perhaps print a nomem error instead of this message?

Done.

> Also, if the strdup fails for Fontname, won’t libass print “Not understood” for each of the following lines? Can we avoid this somehow?

With `ass_nomem_log(lib, false, …)` this would kind of work, **but** assuming the relevant part of the file is being parsed 
before any frame is rendered,  this only works because all other parsing nomem-logs are being forced right now – which might change. If there was any non-forced nomem-log before encountering a mangled font line, no log messages about it would be printed at all.

Reply:
To be clear, what I mean is:

An embedded font usually spans many consecutive lines. Each of those lines results in a call to `process_fonts_line`. The first line contains the font name, and the rest have the font data. If `process_fonts_line` fails to read `fontname` on the first line, and sets `fontname` to `NULL`, then each of the following lines will hit this conditional in `process_fonts_line`:

https://github.com/libass/libass/blob/f6101d16a23879a8f0d220f846a51fcc1ae99976/libass/ass.c#L908-L911

and print a message—not via `ass_nomem_log` but via `ass_msg`!—saying it doesn’t understand the line. This isn’t quite the the point of this PR, but one way or another it’d be great to avoid this flood of messages.

Are you suggesting to just change that `ass_msg` call into `ass_nomem_log(false)`? That’s certainly a (cheap) option. Ideally I suppose this should have its own flag similar to the one for nomem log messages… one that is cleared when the font section is entered.

Reply:
> To be clear, what I mean is:
> -snip-

Yes, that's what I meant. But I don't think there's currently a good way to suppress these message.

> Are you suggesting to just change that ass_msg call into ass_nomem_log(false)? That’s certainly a (cheap) option.

Not really, I brought it up to point out why it is badly suited for this — and without adding a log just for this there's no better option I'm aware of.

> Ideally I suppose this should have its own flag similar to the one for nomem log messages… one that is cleared when the font section is entered.

If a clean log on broken files without an valid fontname line or in case of strdup-failure is important then a special flag just for this would probably be required.

Reply:
Right, I suppose this also happens if a file has a font without a name line.Let's ignore this for now.

(Sigh, why does GitHub allow email replies to these threads but not actually link them to the thread…)